libraries/AP_HAL_ChibiOS/RCOutput.cpp

```diff
@@ -288,7 +288,7 @@ void RCOutput::dshot_collect_dma_locks(uint64_t time_out_us, bool led_thread)
     for (int8_t i = NUM_GROUPS - 1; i >= 0; i--) {
         pwm_group &group = pwm_group_list[i];
 
-        if ((led_thread && !is_led_protocol(group.current_mode)) || is_led_protocol(group.current_mode)) {
+        if (led_thread != is_led_protocol(group.current_mode)) {
```

Comment:
this looks like a non functional change

Reply:
If led_thread is true and is_led_protocol(group.current_mode) is true then we will skip collecting locks, so we are not collecting locks in the led thread